packages/playground/data-liberation/src/entity-readers/WP_WXR_Sorted_Entity_Reader.php

```diff
@@ -0,0 +1,669 @@
+
+	/**
+	 * The base name of the table used to store the IDs, the new IDs and the
+	 * sort order.
+	 */
+	const TABLE_NAME = 'data_liberation_map';
+
+	/**
+	 * The current database version, to be used with dbDelta.
+	 */
+	const DB_VERSION = 1;
+
+	/**
+	 * The current session ID.
+	 */
+	protected $current_session = null;
+
+	/**
+	 * Se to true if the cursors should be read from the map.
+	 */
+	public $emit_cursor = false;
+
+	/**
+	 * The entity types saved in the database.
+	 */
+	const ENTITY_TYPES = array(
+		'category'     => 1,
+		'post'         => 2,
+		'site_option'  => 3,
+		'user'         => 4,
+		'term'         => 5,
+	);
+
+	/**
+	 * The name of the field where the ID is saved.
+	 */
+	const ENTITY_TYPES_ID = array(
+		'category'     => 'slug',
+		'post'         => 'post_id',
+		'site_option'  => 'option_name',
+		'user'         => 'user_login',
+		'term'         => 'term_id',
+	);
+
+	/**
+	 * Create the reader.
+	 *
+	 * @param WP_Byte_Reader $upstream The upstream reader.
+	 * @param mixed         $cursor The cursor.
+	 * @param array         $options The options.
+	 *
+	 * @return WP_WXR_Sorted_Entity_Reader The reader.
+	 */
+	public static function create( WP_Byte_Reader $upstream = null, $cursor = null, $options = array() ) {
+		// Initialize WP_WXR_Entity_Reader.
+		$reader = parent::create( $upstream, $cursor, $options );
+
+		if ( array_key_exists( 'post_id', $options ) ) {
+			// Get the session ID from the post ID.
+			$reader->current_session = $options['post_id'];
+		}
+
+		return $reader;
+	}
+
+	/**
+	 * Advances to the next entity in the WXR file.
+	 *
+	 * @since WP_VERSION
+	 *
+	 * @return bool Whether another entity was found.
+	 */
+	protected function read_next_entity() {
+		if ( ! $this->emit_cursor ) {
+			return parent::read_next_entity();
+		}
+
+		$next_cursor = $this->get_next_cursor();
+
+		if ( ! empty( $next_cursor ) ) {
+			$next_cursor = json_decode( $next_cursor, true );
+
+			/*if ( ! empty( $next_cursor ) ) {
+				$this->last_post_id    = $next_cursor['last_post_id'];
+				$this->last_comment_id = $next_cursor['last_comment_id'];
+				$this->last_term_id    = $next_cursor['last_term_id'];
+				$this->upstream->seek( $next_cursor['upstream'] );
+
+				// Reset the XML processor to the cursor.
+				// $this->xml->reset_to( $next_cursor['xml'] );
+				$this->xml = WP_XML_Processor::create_for_streaming( '', $next_cursor['xml'] );
+				echo "Reset to {$next_cursor['xml']}\n";
+			}*/
+		}
+
+		return parent::read_next_entity();
+	}
+
+	/**
+	 * Get the name of the SQL table.
+	 *
+	 * @return string The name of the table.
+	 */
+	public static function get_table_name() {
+		global $wpdb;
+
+		// Default is wp_{TABLE_NAME}
+		return $wpdb->prefix . self::TABLE_NAME;
+	}
+
+	/**
+	 * Run during the register_activation_hook or similar actions. It creates
+	 * the table if it doesn't exist.
+	 */
+	public static function create_or_update_db() {
```

Comment:
Nit: Let's clarify this is about a table

```suggestion
	public static function create_or_update_db_table() {
```